src/net_processing.cpp

```diff
@@ -2572,7 +2587,8 @@ void PeerManagerImpl::ProcessMessage(CNode& pfrom, const std::string& msg_type,
         UpdatePreferredDownload(pfrom, State(pfrom.GetId()));
         }
 
-        if (!pfrom.IsInboundConn() && !pfrom.IsBlockOnlyConn()) {
+        // Self advertisement logic
```

Comment:
This is a bit misleading. The logic block below contains self-advertisement logic, but it also sends a `getaddr` to request addresses, which is not self-advertisement.

Reply:
good point, updated to include